ddtrace/llmobs/_writer.py

```diff
@@ -507,11 +514,18 @@ def dataset_get_with_records(self, dataset_name: str, project_name: Optional[str
             has_next_page = False
             if next_cursor:
                 has_next_page = True
-                list_path = f"{list_base_path}?page[cursor]={next_cursor}"
+                list_path = f"{list_base_path}{'&' if version else '?'}page[cursor]={next_cursor}"
                 logger.debug("next list records request path %s", list_path)
                 page_num += 1
         return Dataset(
-            dataset_name, project, dataset_id, class_records, dataset_description, curr_version, _dne_client=self
+            dataset_name,
+            project,
+            dataset_id,
+            class_records,
+            dataset_description,
+            curr_version,
+            version or curr_version,
+            _dne_client=self,
```

Comment:
similar small nit for maybe using kwargs to give each of these a name for slightly easier readability - up to you!